godot-core/src/registry/property.rs

```diff
@@ -77,6 +77,9 @@ pub trait Export: Var {
     }
 }
 
+/// Marker trait to identify non-nullable `GodotType`s that can be directly used with `#[export]` and `#[var]`.
+pub trait DirectExport {}
```

Comment:
Great idea to address the "too many impls" problem, thanks! :+1: 

I'm not sure if the term "direct" is very precise. What about `BuiltinExport` -- in practice, all types for which the trait is implemented are builtins, no?

An alternative would also be to check this via `Bounds` associated type -- like the `Exportable` type here, it could be something like `BuiltinExportable` or so.
https://github.com/godot-rust/gdext/blob/027a9690a7eb2544fdbb13dd9d8e7426d1a49abc/godot-core/src/obj/bounds.rs#L100-L104

These are all just ideas, I'm fine with getting the basics sorted in this PR and then refine later :slightly_smiling_face:

Reply:
I tried it via Bounds and it seems to work – therefore associa `DirectExport` is now a part of the GodotType as an associated type `BuiltinExportable: Exportable`; 

The question is if we want to move declaration of `Exportable` and `bounds::Yes/No`, for now on I left it in `obj/bounds.rs` :shrug:..

Earlier on I couldn't get it working :thinking:  (probably because I haven't specified both `OnEditor<T>: GodotConvert<Via = T>` and `T: GodotConvert<Via = T>` in required bounds).

Reply:
Sorry, I meant to use an associated type inside `Bounds`, not `GodotType`. But I missed that `Bounds` doesn't support non-class types :man_facepalming: But I don't think `GodotType` is a good idea, this is a very core type that almost everything must implement, so adding more requirements there has far-reaching consequences.

Probably the original proposal of yours was indeed better. There are some marker traits in [`godot::meta`](https://godot-rust.github.io/docs/gdext/master/godot/meta/index.html) already; we could possibly add `BuiltinExportable` there?

Also, where does `Variant` stand in this? Like object, it has a null state (`Variant::nil()`), should `OnEditor` expect that it's non-nil as well?